starter-code/pizza.js

```diff
@@ -45,26 +45,105 @@ function renderPepperonni() {
 
 function renderMushrooms() {
   // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll('.mushroom').forEach(function($mushroom){
+    if (state.mushrooms) {
+      $mushroom.style.visibility = "visible";
+    }
+    else {
+      $mushroom.style.visibility ="hidden";
+    }
+  })
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll('.green-pepper').forEach(function($greenPepper){
+    if (state.greenPeppers) {
+      $greenPepper.style.visibility = "visible";
+    }
+    else {
+      $greenPepper.style.visibility ="hidden";
+    }
+  })
 }
 
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
-}
+  const SW = document.querySelector('.sauce');
+    if (state.whiteSauce) {
+      SW.classList.add('sauce-white');
+    }
+    else {
+      SW.classList.remove('sauce-white');
+    }
+    // SW.classList.toggle("sauce-white");
+  }
+
 
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+  const FC = document.querySelector('.crust');
+    if (state.glutenFreeCrust) {
+      FC.classList.add('crust-gluten-free');
+    }
+    else {
+      FC.classList.remove('crust-gluten-free');
+    }
 }
 
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  const pepButton = document.querySelector('.btn.btn-pepperonni');
+  if (state.pepperonni) {
+    pepButton.classList.add('active');
+   } 
+   else {
+    pepButton.classList.remove('active');
+  }
+  
+  const mushButton = document.querySelector('.btn.btn-mushrooms');
+  if (state.mushrooms) {
+    mushButton.classList.add('active');
+  }
+  else {mushButton.classList.remove('active');
+}
+  
+  const greenButton = document.querySelector('.btn.btn-green-peppers');
+  if (state.greenPeppers) {
+    greenButton.classList.add('active');
+  }
+  else {greenButton.classList.remove('active');
+}
+  
+  const sauceButton = document.querySelector('.btn.btn-sauce');
+  if (state.whiteSauce) {
+    sauceButton.classList.add('active');
+  }
+  else {sauceButton.classList.remove('active');
+}
+  
+  const crustButton = document.querySelector('.btn.btn-crust');
+  if (state.glutenFreeCrust) {
+    crustButton.classList.add('active');
+  }
+  else {crustButton.classList.remove('active');
+}
 }
 
+
 function renderPrice() {
   // Iteration 4: change the HTML of `<aside class="panel price">`
+  var endPrice = basePrice;
+  var list = document.querySelector("aside.panel.price ul");
+  list.innerHTML = "";
+
+  for (let i in ingredients) {
+    if (state[i]) {
+      endPrice += ingredients[i].price;
+      list.innerHTML += `<li>${ingredients[i].price} ${ingredients[i].name}</li>`
+    }
+  }
+  document.querySelector("aside.panel.price strong").innerHTML = "$" + endPrice;
```

Comment:
It would be better if you used "key" or "prop" as a variable when iterating over objects since it's more descriptive about what it is.